src/lib/routes.ts

```diff
@@ -37,14 +38,40 @@ const routesOrder: ReadonlyArray<Route> = [
 const isEditing = () =>
   typeof window !== 'undefined' && window.location.search.includes('edit')
 
-export const getRoutes = (currentRoute: Route) => {
+export const getRoutes = (currentRoute: Route, taxForm: TaxForm) => {
   const currentRouteIndex = routesOrder.indexOf(currentRoute)
   return {
     currentRoute,
-    nextRoute: () =>
-      isEditing() ? '/suhrn' : routesOrder[currentRouteIndex + 1],
-    previousRoute: () =>
-      isEditing() ? '/suhrn' : routesOrder[currentRouteIndex + -1],
+    nextRoute: () => {
+      if (isEditing()) {
+        return '/suhrn'
+      } else if (currentRoute === '/partner') {
```

Comment:
co keby sme tuto radsej vyfiltrovali `routesOrder` podla `eligibility`? Napriklad ak `!taxForm.eligibleForChildrenBonus` tak dame prec `/deti` z `routesOrder`. Potom stale mozeme vzdy pouzivat +1 pre next route. Ked je tam +2 tak tam chyba explicitne ze ktoru routu preskakujes (preto tam je teraz potrebny ten comment)

Vies ako to myslim?

Zda sa mi to cistejsie a skalovatelnejsie.

Reply:
Good point